pkgs/development/tools/build-managers/gradle/fetch-deps.nix

```diff
@@ -123,10 +123,15 @@ let
 
             fileList = builtins.filter (x: lib.hasPrefix xmlBase x && x != url) (builtins.attrNames finalData);
             jarPomList = map parseArtifactUrl fileList;
-            sortedJarPomList =
-              lib.sort
-                (a: b: lib.splitVersion a.version < lib.splitVersion b.version)
-                jarPomList;
+
+            a-before-b = a: b:
+              let as      = lib.splitVersion a.version;
+                  bs      = lib.splitVersion b.version;
+                  n       = lib.min (lib.length as) (lib.length bs);
```

Comment:
how will this deal with comparing e.g. `2.3.5` and `2.3`?

Reply:
Yeah, I'm off the mark here. This doesn't do the comparison right. In 2.3 and continuing to the present Nix version there's `builtins.compareVersion`. I didn't follow your PR closely enough to figure out whether that had semantics you wanted or not.

Reply:
since I need the versions sorted, it has to use `builtins.compareVersion` or an exact alternative